packages/core/src/plugins/PluginManager.ts

```diff
@@ -205,13 +205,13 @@ export class PluginManager {
       .reduce((renderedContent, plugin) => plugin.postRender(frontmatter, renderedContent), content);
   }
 
-  processNode(node: DomElement) {
+  processNode(node: NodeOrText) {
```

Comment:
Just curious. Seems like in some places, `DomElement` was replaced by `MbNode` and in other places `DomElement` was replaced by `NodeOrTest`. How do you decide which to replace `DomElement` by?

Reply:
@raysonkoh MbNode is a stricter type than NodeOrText, so I try to use MbNode wherever possible so that we can assert that the node contains some attributes using types instead of using the code logic (which is repetitive since the linter would enforce that we do so repeatedly). NodeOrText is used when we cannot guarantee what kind of node it is - i.e. it may be a TextElement, which is simpler/looser than MbNode and doesn't contain the same assertions. 
May be best illustrated with the example from parseHTML:
```typescript
export function parseHTML(html: string) {
  const node = cheerio.parseHTML(html);
  return node as unknown as NodeOrText[];
}
```
NodeOrText is used as we don't know what HTML string this method will be called with, which means that the parsed node might be a TextElement or an MbNode. Using NodeOrText allows us to do a typecast nearer the caller when we can guarantee that the node will be a MbNode:
```typescript
  const test = '<panel><div #header>test</div></panel>';
  const testNode = parseHTML(test)[0] as MbNode;
```
We can typecast here as we know that `test = '<panel><div #header>test</div></panel>'` is a node with a name, attributes, etc, and not a TextElement.

Reply:
Got it, thanks! I think it would be good to document this somewhere in the devdocs otherwise people might use `NodeOrText` or `MbNode` in inappropriate places.

Reply:
Much of this is summarised in [node.ts](https://github.com/jovyntls/markbind/blob/ts-unify-node-types/packages/core/src/utils/node.ts); do you think it is sufficient, or it would be helpful to add the above example to the devdocs as well? @raysonkoh